### ISSUE TITLE
Add Fast Refresh demo to examples and docs

### ISSUE THREAD

**Issue by lfades** [vercel/next.js#16538]
Add a new example called `fast-refresh` to our examples folder that showcases the different features and behaviors of [Fast Refresh](https://nextjs.org/docs/basic-features/fast-refresh).

Use this removed example for inspiration: https://github.com/vercel/next.js/pull/16497 

Once the example is ready, add a link to it in our docs.

**Comment by YichiZ** [comment]
Going to try this out...

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
